rest/src/test/java/org/opentestsystem/authoring/testspecbank/integration/TestSpecificationServiceTest.java

```diff
@@ -534,6 +545,49 @@ public void retryExportPackageForRetiredSpec() {
         }
     }
 
+    @Test
+    public void shouldDeleteATestSpecification() {
+        final TestSpecification testSpecificationToDelete = PODAM_FACTORY.manufacturePojo(TestSpecification.class);
+        testSpecificationToDelete.setId(null);
+        testSpecificationToDelete.setLastUpdatedDate(null);
+        testSpecificationToDelete.setSpecificationXml(TestSpecificationXmlBuilder.generateXml(TestSpecificationXmlBuilder.buildValidTestspecification(Purpose.SCORING)));
+        final TestSpecification savedTestSpecificationToDelete = this.testSpecificationService.saveTestSpecification(testSpecificationToDelete);
+        assertThat(savedTestSpecificationToDelete, is(notNullValue()));
+
+        final TestSpecification testSpecificationToRemain = PODAM_FACTORY.manufacturePojo(TestSpecification.class);
+        testSpecificationToRemain.setId(null);
+        testSpecificationToRemain.setLastUpdatedDate(null);
+        testSpecificationToRemain.setSpecificationXml(TestSpecificationXmlBuilder.generateXml(TestSpecificationXmlBuilder.buildValidTestspecification(Purpose.SCORING)));
+        testSpecificationRepository.save(testSpecificationToRemain);
+
+        testSpecificationService.deleteTestSpecification(testSpecificationToDelete.getName());
+
+        final TestSpecification foundSpecification = testSpecificationRepository.findOneByName(testSpecificationToDelete.getName());
+        assertThat(foundSpecification, is(nullValue()));
+        final GridFSDBFile foundGridFsFile = gridFsRepository.getById(testSpecificationToDelete.getSpecificationXmlGridFsId());
+        assertThat(foundGridFsFile, is(nullValue()));
+
+        final TestSpecification foundSpecificationThatShouldRemain = testSpecificationRepository.findOneByName(testSpecificationToRemain.getName());
+        assertThat(foundSpecificationThatShouldRemain, is(not(nullValue())));
```

Comment:
kind of weird that we don't make sure we're deleting the right one, but I guess we are trusting the findOneByName to do the right thing and give us the right one (or null). glad you're making sure it doesn't do a 'delete all'.